common/ensemble_building/singlebest_ensemble.py

```diff
@@ -0,0 +1,137 @@
+import os
+from typing import List, Tuple, Union, cast
+
+import numpy as np
+
+from sklearn.pipeline import Pipeline
+
+from smac.runhistory.runhistory import RunHistory
+
+from .abstract_ensemble import AbstractEnsemble
+from ..utils.backend import Backend
+
+
+class SingleBest(AbstractEnsemble):
+    """
+    In the case of a crash, this class searches
+    for the best individual model.
+
+    Such model is returned as an ensemble of a single
+    object, to comply with the expected interface of an
+    AbstractEnsemble.
+    """
+
+    def __init__(
+        self,
+        run_history: RunHistory,
+        seed: int,
+        backend: Backend,
+    ):
+        self.seed = seed
+        self.backend = backend
+
+        # Add some default values -- at least 1 model in ensemble is assumed
+        self.indices_ = [0]
+        self.weights_ = [1.0]
+        self.run_history = run_history
+        self.best_loss = np.inf
+        self.identifiers_ = self.get_identifiers_from_run_history()
+
+    def fit(
+        self,
+        base_models_predictions: np.ndarray,
+        true_targets: np.ndarray,
+        model_identifiers: List[Tuple[int, int, float]],
+    ) -> "AbstractEnsemble":
+        return self
+
+    def get_identifiers_from_run_history(self) -> List[Tuple[int, int, float]]:
```

Comment:
But there are plenty of this to check, like if the path exists (it might have been deleted by ensemble builder), among others. This suggestion does not align to that. Yes, one could do list comprehension with filtering, but then it is not much different than the actual code...

Reply:
I will check here later (my time schedule is temporally tight..., I think I can check on this weekend)